Add Internal AI Evaluation

laHermes opened the pull request:
### Features
- **Internal AI Evaluation**: Adds a reviewer-only evaluation that is not visible to applicants
  - New `InternalAIEvaluationDisplay` component with lock icon indicator
  - Separate API endpoint (`runInternalAIEvaluation`) for internal evaluations
  - Configurable via `internalLangfusePromptId` in program AI settings
  - Manual trigger button for admins (disabled if prompt not configured)

### Refactoring
- **Extracted Shared Components**: Created `EvaluationComponents.tsx` to eliminate ~500 lines of duplicate code
  - Shared components: `ScoreDisplay`, `DecisionDisplay`, `DisqualificationReason`, `EvaluationSummary`, `ImprovementRecommendations`, `AdditionalNotes`, `StatusChip`, `EvaluationDisplay`
  - Both `AIEvaluation.tsx` and `InternalAIEvaluation.tsx` now use shared components
  - Customizable footer disclaimer via `footerDisclaimer` prop

### Changes
- Added `internalLangfusePromptId` to form schema and program config types
- Updated `AIEvaluationButton` to support both regular and internal evaluation modes
- Added warning UI when internal prompt is not configured
- Updated admin and reviewer pages to show internal evaluation section

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Introduces reviewer-only internal AI evaluation with a manual trigger, refactors evaluation UI into shared components, adds config/types support, and wires it into admin/reviewer views.
> 
> - **Funding Platform UI**:
>   - **Internal Evaluation**: New `InternalAIEvaluationDisplay` and section in `ApplicationContent` with gated button, pending state, and warning when `internalLangfusePromptId` is missing.
>   - **Evaluation Trigger**: `AIEvaluationButton` supports `isInternal`, improved loading, toasts, accessibility, and error handling; calls correct API.
>   - **Refactor**: Extract shared UI into `EvaluationComponents` and helpers in `evaluationUtils`; `AIEvaluation` updated to use them.
>   - **Pages**: Admin/Reviewer pages enable `showInternalEvaluation` and pass through props.
>   - **Config UI**: `AIPromptConfiguration` adds `internalLangfusePromptId` field with live schema updates.
>   - **Utility**: Add `LoadingSpinner` component.
> - **Services**:
>   - **API**: Add `fundingApplicationsAPI.runInternalAIEvaluation` (POST `/evaluate-internal`, timeout), and harden export filename parsing.
> - **Types**:
>   - Extend `IAIConfig` with `internalLangfusePromptId` and `IFundingApplication` with `internalAIEvaluation`; add `ProgramWithFormSchema` helper type.
> - **Tests**:
>   - Comprehensive tests for button, content integration, shared components, and utils.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit df1f58ce162a2ad849e7bac19eb282d34d4636ad. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

laHermes committed:
refactor: enhance reviewer application detail page and AI evaluation functionality

laHermes committed:
fix: improve error handling and user feedback for AI evaluation button

laHermes committed:
test: add comprehensive unit tests for AI evaluation components

laHermes committed:
fix: increase timeout for funding application evaluation API call

laHermes committed:
Merge remote-tracking branch 'origin' into feat/internal-ai-evaluation

laHermes committed:
fix: improve error handling in AIEvaluationButton component

amaurymagalhaes reviewed (commented):
Great PR! Here are some minor suggestions for improvement:

amaurymagalhaes reviewed (commented):
Nice work on the refactoring! Here are some minor suggestions for improvement:

amaurymagalhaes commented on components/FundingPlatform/ApplicationView/EvaluationComponents.tsx:
```diff
@@ -0,0 +1,486 @@
+}: EvaluationDisplayProps) {
+  // Check if program is audit grants or growth grants
+  const isAuditGrants = false;
+  const isGrowthGrants = false;
```
**Suggestion**: These flags are always hardcoded to `false`:
```typescript
const isAuditGrants = false;
const isGrowthGrants = false;
```

Consider either:
1. Passing them as props to `EvaluationDisplay` if they should be configurable per-program
2. Adding a comment explaining why they're always `false` (e.g., "// Feature disabled pending product decision")
3. Removing the conditional logic if these features are deprecated

Not a blocker, just wanted to flag for clarity! 🔍

amaurymagalhaes committed:
fix: address PR review comments

- Remove unused isAxiosErrorWithResponse type guard in AIEvaluationButton.tsx
- Add comments explaining hardcoded isAuditGrants/isGrowthGrants flags
- Refactor IIFE pattern to useMemo for renderedFields tracking
- Remove unused isLoading prop from InternalAIEvaluationDisplay
- Extract ProgramWithFormSchema type to shared types file
